### ISSUE TITLE
`lotus-storage-miner storage attach` takes a very long time

### ISSUE THREAD

**Issue by whyrusleeping** [filecoin-project/lotus#1793]
Not exactly sure what its doing, but adding new storage to the system takes an annoyingly long time (I assume its waiting on a lock). I have a sector in precommit1 right now, so maybe it will block until that is done?

**Comment by whyrusleeping** [comment]
stack trace: https://ipfs.io/ipfs/QmcC7ee63qAufFsKVGeX3T6j8tQZVjmVJbrBydmej2jks8

**Comment by magik6k** [comment]
```
goroutine 4196692 [semacquire, 6 minutes]:
sync.runtime_SemacquireMutex(0xc001ead0c8, 0xc002ec0200, 0x0)
	/usr/local/lib/go/src/runtime/sema.go:71 +0x47
sync.(*RWMutex).Lock(0xc001ead0c0)
	/usr/local/lib/go/src/sync/rwmutex.go:103 +0x88
github.com/filecoin-project/sector-storage/stores.(*Local).OpenPath(0xc001ead080, 0x279d6a0, 0xc0040ffe00, 0xc0022efea0, 0x17, 0x0, 0x0)
	/home/why/go/pkg/mod/github.com/filecoin-project/sector-storage@v0.0.0-20200515123304-20817dc51db5/stores/local.go:82 +0x6f
github.com/filecoin-project/sector-storage.(*Manager).AddLocalStorage(0xc001ead380, 0x279d6a0, 0xc0040ffe00, 0xc0022efea0, 0x17, 0xc0035631f8, 0x51e6e3)
	/home/why/go/pkg/mod/github.com/filecoin-project/sector-storage@v0.0.0-20200515123304-20817dc51db5/manager.go:139 +0x102
github.com/filecoin-project/lotus/node/impl.(*StorageMinerAPI).StorageAddLocal(0xc000194cf0, 0x279d6a0, 0xc0040ffe00, 0xc0022efea0, 0x17, 0x0, 0x0)
	/home/why/code/lotus/node/impl/storminer.go:223 +0x60
reflect.Value.call(0x23ecba0, 0xc000194cf0, 0x6e13, 0x240b977, 0x4, 0xc0040ffef0, 0x2, 0x2, 0x5, 0x211f001, ...)
	/usr/local/lib/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x23ecba0, 0xc000194cf0, 0x6e13, 0xc0040ffef0, 0x2, 0x2, 0x517678, 0x8, 0x10)
	/usr/local/lib/go/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/lotus/api/apistruct.permissionedAny.func1(0xc0040ffef0, 0x2, 0x2, 0xc0040ffef0, 0x21736c0, 0xc000449501)
	/home/why/code/lotus/api/apistruct/permissioned.go:94 +0x510
reflect.callReflect(0xc0020ddef0, 0xc003563708, 0xc0035636f0)
	/usr/local/lib/go/src/reflect/value.go:549 +0x322
reflect.makeFuncStub(0x279d6a0, 0xc0040ffe00, 0xc0022efea0, 0x17, 0xc003563760, 0x51e6e3, 0xc003563788, 0x5701ab, 0xc00015b680, 0x279d6a0, ...)
	/usr/local/lib/go/src/reflect/asm_amd64.s:20 +0x42
github.com/filecoin-project/lotus/api/apistruct.(*StorageMinerStruct).StorageAddLocal(0xc00015b680, 0x279d6a0, 0xc0040ffe00, 0xc0022efea0, 0x17, 0x0, 0x0)
	/home/why/code/lotus/api/apistruct/struct.go:809 +0x55
reflect.Value.call(0xc002d87620, 0xc000639848, 0x13, 0x240b977, 0x4, 0xc005b2ae10, 0x3, 0x3, 0x51e79f, 0x21401e0, ...)
	/usr/local/lib/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0xc002d87620, 0xc000639848, 0x13, 0xc005b2ae10, 0x3, 0x3, 0x21401e0, 0xc005f54d90, 0x198)
	/usr/local/lib/go/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/lotus/lib/jsonrpc.doCall(0xc003724140, 0x18, 0xc002d87620, 0xc000639848, 0x13, 0xc005b2ae10, 0x3, 0x3, 0x0, 0x0, ...)
	/home/why/code/lotus/lib/jsonrpc/handler.go:129 +0xdc
github.com/filecoin-project/lotus/lib/jsonrpc.handlers.handle(0xc0020dcc90, 0x279d6a0, 0xc0040ffda0, 0xc006800ae0, 0x3, 0xc006800ae8, 0xc003724140, 0x18, 0xc002608480, 0x1, ...)
	/home/why/code/lotus/lib/jsonrpc/handler.go:205 +0xb2b
created by github.com/filecoin-project/lotus/lib/jsonrpc.(*wsConn).handleCall
	/home/why/code/lotus/lib/jsonrpc/websocket.go:407 +0x295
```

**Comment by chengwds** [comment]
`lotus-storage-miner storage attach` takes a very long time，That's what it takes

**Comment by magik6k** [comment]
Fixed in #1924
